src/app/application/components/data-source/DataSource.vue

```diff
@@ -0,0 +1,84 @@
+<script setup lang="ts">
+
+import { watch, ref, onBeforeUnmount } from 'vue'
+
+import { useDataSourcePool } from '@/utilities'
+
+const data = useDataSourcePool()
+
+const props = defineProps({
+  src: {
+    type: String,
+    required: true,
+  },
+})
+
+const message = ref<unknown>(undefined)
+const error = ref<Error | undefined>(undefined)
+
+const emit = defineEmits<{
+  (e: 'change', value: any): void
+  (e: 'error', error: Error): void
+}>()
+
+type State = {
+  controller?: AbortController
+  src?: string
+}
+let state: State = {}
+const sym = Symbol('')
+const open = async (src: string) => {
+  state = close(state)
+  state.src = src
+  if (src === '') {
+    return
+  }
+  state.controller = new AbortController()
+  // this should emit proper events
+  const source = data.source(src, sym)
```

Comment:
Maybe this is `acquire`? Or `getOrCreate`

Reply:
I wanted it to read like `data source` ie. "give me a data source please, I don't care how just go get it for me".

The fact that there is a pool underneath is purposefully hidden from the interface of the datasource itself, hence why I much prefer `data.source` rather than things like `acquire`. Then `source.open` and `source.close` which sticks closer to the semantics of `EventSource` which is what this is modelled on.